test/dtypes/test_fpx.py

```diff
@@ -69,61 +73,41 @@ def test_from_scaled_tc_fpx_compile(self, ebits, mbits, device):
     @pytest.mark.skipif(not torch.cuda.is_available(), reason="CUDA not available")
     @parametrize("ebits,mbits", _FPx_DTYPES)
     def test_to_copy_device(self, ebits, mbits):
-        x = torch.randn(256, 64)
-        fpx = QuantLlmLinearWeight.from_float(x, ebits, mbits).cuda()
-        assert fpx.device.type == "cuda"
-        fpx = fpx.cpu()
-        assert fpx.device.type == "cpu"
+        from torchao.quantization.quant_primitives import (
+            choose_qparams_affine_fpx,
+            quantize_affine_fpx,
+        )
 
-    @pytest.mark.skipif(not torch.cuda.is_available(), reason="CUDA not available")
-    @parametrize("ebits,mbits", _FPx_DTYPES)
-    @parametrize("leading_dims", [(4,), (2, 4)])
-    @parametrize("bias", [False, True])
-    def test_quant_llm_linear_weight(self, ebits, mbits, bias, leading_dims):
-        OC, IC = 256, 64
-        device = "cuda"
-
-        fp16_weight = torch.randn(OC, IC, device=device, dtype=torch.half)
-        fp16_bias = torch.randn(OC, device=device, dtype=torch.half) if bias else None
-
-        fpx_weight = QuantLlmLinearWeight.from_float(fp16_weight, ebits, mbits)
-
-        x = torch.randn(*leading_dims, IC, device=device, dtype=torch.half)
-        out = torch.nn.functional.linear(x, fpx_weight, fp16_bias)
-        assert out.shape == leading_dims + (OC,)
+        x = torch.randn(256, 64)
+        scale = choose_qparams_affine_fpx(x, ebits, mbits)
+        x = quantize_affine_fpx(x, scale, ebits, mbits)
+        layout_type = FpxTensorCoreLayoutType(ebits, mbits)
+        fpx_layout_tensor = FpxTensorCoreAQTLayout.from_plain(x, scale, None, layout_type).cuda()
+        assert fpx_layout_tensor.device.type == "cuda"
+        fpx_layout_tensor = fpx_layout_tensor.cpu()
+        assert fpx_layout_tensor.device.type == "cpu"
 
     @pytest.mark.skipif(not torch.cuda.is_available(), reason="CUDA not available")
+    @pytest.mark.skipif(not TORCH_VERSION_AT_LEAST_2_5, reason="quantization only works with torch.compile for 2.5+")
```

Comment:
Just curious, what is the reason that the new change doesn't work with previous versions of PyTorch anymore? Don't think anyone is actively using this, so probably not too important to avoid BC.

Reply:
this is because after integrating this into AffineQuantizedTensor, we now have multiple levels of tensor subclass, and it does not work with torch.compile until we upgrade pytorch to after 2.5